fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaViewModel.java

```diff
@@ -214,18 +222,17 @@ public void addDisposable(Disposable disposable) {
     this.disposables.add(disposable);
   }
 
+  public LaoView getLaoView(String laoId) throws UnknownLaoException {
+    return laoRepository.getLaoView(laoId);
+  }
+
   @Nullable
-  public Lao getCurrentLao() {
+  public Lao getCurrentLao() throws UnknownLaoException {
     return getLao(getLaoId().getValue());
   }
 
   @Nullable
-  private Lao getLao(String laoId) {
-    // TODO Fully move to an LaoView here (and throw the exception further)
-    try {
-      return laoRepository.getLaoView(laoId).createLaoCopy();
-    } catch (UnknownLaoException e) {
-      return null;
-    }
+  private Lao getLao(String laoId) throws UnknownLaoException {
+    return laoRepository.getLaoView(laoId).createLaoCopy();
```

Comment:
What is stopping us from moving to an LaoView ?

Reply:
To make it clean it requires quite some changes in the codebase. Those changes make sense since it's a complete transition to LaoView in all part of the codebase except the handlers and laoRepo. It's a work in progress in a separate branch

Reply:
👌